tests/test_events.py

```diff
@@ -17,119 +17,136 @@
     dataset = None
     for _ in range(RETRIES):
         await asyncio.sleep(RETRY_INTERVAL)
         try:
-            dataset = await repository.get_dataset(accession)
+            dataset = await repository.get_dataset("some-dataset-id")
         except repository.DatasetNotFoundError:
             pass
         else:
-            assert dataset.title == "Test dataset 1"
+            assert dataset == DATASET
             break
     else:
-        assert False, "dataset not created"
+        assert False, "dataset cannot be retrieved"
+
+    # but another dataset should not be retrievable
+    with pytest.raises(repository.DatasetNotFoundError):
+        await repository.get_dataset("another-dataset-id")
+
+
+async def test_dataset_update(
+    config: Config, kafka_fixture: KafkaFixture, consumer: Consumer, populated_mongodb
+):
+    """Test updating a dataset via an event."""
+    repository, subscriber = consumer
+
+    # make sure that in the beginning the dataset exists
+    dataset = await repository.get_dataset("some-dataset-id")
+    assert dataset.id == "some-dataset-id"
+    assert dataset.title == "Test dataset 1"
 
     # update the dataset
 
     updated_dataset = DATASET_UPSERTION_EVENT
-    updated_dataset = inserted_dataset.model_copy(
-        update={"accession": accession, "title": "Changed dataset 1"}
-    )
+    updated_dataset = updated_dataset.model_copy(update={"title": "Changed dataset 1"})
     await kafka_fixture.publish_event(
         payload=updated_dataset.model_dump(),
         topic=config.dataset_change_event_topic,
         type_=config.dataset_upsertion_event_type,
-        key=key,
+        key="test-key",
     )
-    await asyncio.wait_for(run(forever=False), timeout=TIMEOUT)
+    await asyncio.wait_for(subscriber.run(forever=False), timeout=TIMEOUT)
     # wait until dataset is updated
-    dataset = None
     for _ in range(RETRIES):
         await asyncio.sleep(RETRY_INTERVAL)
-        dataset = await repository.get_dataset(accession)
+        dataset = await repository.get_dataset(dataset.id)
         if dataset.title == "Changed dataset 1":
             break
     else:
         assert False, "dataset title not changed"
 
+
+async def test_dataset_deletion(
+    config: Config, kafka_fixture: KafkaFixture, consumer: Consumer, populated_mongodb
+):
+    """Test deleting a dataset via an event."""
+    repository, subscriber = consumer
+
+    # make sure that in the beginning the dataset exists
+    dataset = await repository.get_dataset("some-dataset-id")
+    assert dataset.id == "some-dataset-id"
+
     # delete the dataset again
 
     deleted_dataset = DATASET_DELETION_EVENT
-    deleted_dataset = deleted_dataset.model_copy(update={"accession": accession})
     await kafka_fixture.publish_event(
         payload=deleted_dataset.model_dump(),
         topic=config.dataset_change_event_topic,
         type_=config.dataset_deletion_event_type,
-        key=key,
+        key="test_key",
```

Comment:
```suggestion
        key="test-key",
```
For consistency with the with the other events.